frontend/src/shared/api.ts

```diff
@@ -16,6 +16,10 @@ export interface Video {
   processing: boolean;
 }
 
+export interface Progress {
+  clips: Record<string, number>
```

Comment:
I wonder if it would be good to convert this into a type, wdyt?

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-aliases